projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -876,9 +891,9 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
         if ((this._ngControl.control.touched || this._ngControl.control.dirty) &&
```

Comment:
We should now check if the control is touched or dirty OR it has a validator or async validator. This is because `onStatusChanged` will be invoked right after we update `disabledDates` and the current implementation expects both for the control to be in a specific state AND to have validators. While this was working just fine, now that we've made our DatePicker into a validator, we have to make sure that its validity is updated no matter if it has been touched or not, since you can easily change the `disabledDates` to be outside of a specific range and the control should become invalid.

Reply:
Fixed, though a bit differently,
as this suggestion is not entirely true. It will force the component to set IgxInputState when there is no user interaction. For example, this will cause setting IgxInputState.**INVALID** and the corresponding RED-ish styles when there is error.required, and we do not want that.

Reply:
done